src/wallets/TransactionHistory.tsx

```diff
@@ -8,18 +8,39 @@ import {ReceiveTransaction} from './modals/ReceiveTransaction'
 import {SortBy, TransactionList} from './TransactionList'
 import {Trans} from '../common/Trans'
 import {asWei, Wei} from '../common/units'
-import {Account, FeeEstimates} from '../common/wallet-state'
+import {Account, FeeEstimates, SynchronizationStatus} from '../common/wallet-state'
+import {useTranslation} from '../settings-state'
 import {Transaction} from './history'
 
 import './TransactionHistory.scss'
 
+const OUT_OF_SYNC_BLOCKS_THRESHOLD = 5
+
+const ConditionalTooltip = ({
+  title,
+  visible,
+  children,
+}: {
+  title: string
+  visible: boolean
```

Comment:
Does `visible` override the default hover behavior?
I'm a little worried that the user fills in all the tx details, then only realizes he can't send the tx when he tries to press submit.

Reply:
Regarding what to do when user already filled tx details:

- Tx won't fail just because of wallet is out of sync. It might fail because wallet does not know correct balance, because it is out of sync (...or nonce is incorrect, but that should not depend on sync status)
- Since user was able to enter send tx flow, the wallet was synced in the first place (and balance was up to date)
- The only case when tx could unexpectedly fail is then if user did some another outgoing tx while being in send tx flow and at the same time, the wallet didn't sync those last blocks.

All in all, I don't feel it's worth to separately handle that case. This whole "disable send tx while syncing" is mostly for the case when user is doing initial sync (or subsequent ones after client was closed).